mllib/src/main/scala/org/apache/spark/ml/clustering/KMeans.scala

```diff
@@ -196,6 +220,14 @@ object KMeansModel extends MLReadable[KMeansModel] {
       val model = new KMeansModel(metadata.uid, new MLlibKMeansModel(clusterCenters))
 
       DefaultParamsReader.getAndSetParams(model, metadata)
+
+      val hasInitialModel = (metadata.metadata \ "hasInitialModel").extract[Boolean]
```

Comment:
It would be nice to maintain backwards compatibility, though we have not promised it yet.  Could you modify this to work with the previous model format (without this field), where you check the saved spark version?

Same for saving the Estimator KMeans